src/main.rs

```diff
@@ -305,68 +300,78 @@ async fn handle_web_req(
+            //  and broadcast the proposed PSBT, or some other error occurs
+            let headers = Headers(req.headers().clone());
+            let query = req.uri().query().map(ToString::to_string);
+            let body_bytes = dbg!(hyper::body::to_bytes(req.into_body()).await?);
+            let proposal = UncheckedProposal::from_request(&*body_bytes, query.as_deref(), headers)
+                .expect("received invalid proposal - TODO: handle this error properly");
             if proposal.is_output_substitution_disabled() {
-                panic!("Output substitution must be enabled");
+                panic!("Output substitution must be enabled"); // TODO: Proper checks for `pjos`
             }
-
             let proposal = proposal
                 .assume_interactive_receive_endpoint() // TODO Check
                 .assume_no_inputs_owned() // TODO Check
                 .assume_no_mixed_input_scripts() // This check is silly and could be ignored
                 .assume_no_inputs_seen_before(); // TODO
 
+            // `psbt` is going to ACTUALLY be the proposal PSBT
             let mut psbt = proposal.psbt().clone();
-            eprintln!("Received transaction: {:#?}", psbt);
-            {
-                for input in &mut psbt.unsigned_tx.input {
-                    // clear signature
-                    input.script_sig = bitcoin::blockdata::script::Script::new();
-                }
-            }
+            eprintln!("Received transaction: {:#?}", psbt); // but not yet
+            psbt.unsigned_tx
+                .input
+                .iter_mut()
+                .for_each(|txin| txin.script_sig = bitcoin::Script::new());
+
+            // find our output
+            // TODO: Would it be possible that the user sends to multiple outputs that we own?
             let (our_output, scheduled_payjoin) = handler
                 .payjoins
                 .find(&mut psbt.unsigned_tx.output)
                 .expect("the transaction doesn't contain our output");
-            let total_channel_amount: bitcoin::Amount = scheduled_payjoin
+
+            // TODO: make this a member fn of `ScheduledPayJoin`
+            // TODO: make sure it is impossible to create a BIP21pj linked to `ScheduledPayJoin`
+            //  with a "total channel amount" larger than requested + fee
+            let total_channel_amount = scheduled_payjoin
                 .channels
                 .iter()
                 .map(|channel| channel.amount)
                 .fold(bitcoin::Amount::ZERO, std::ops::Add::add);
+
+            // TODO: make this a member fn of `ScheduledPayJoin`
             let fees = calculate_fees(
                 scheduled_payjoin.channels.len() as u64,
                 scheduled_payjoin.fee_rate,
                 scheduled_payjoin.wallet_amount != bitcoin::Amount::ZERO,
             );
 
+            // TODO: not great
```

Comment:
make TODOs actionable. Explain qualitative disagreements.

Reply:
I mean we shouldn't have anything that panics when handling an http request